### DISCUSSION TITLE
How to change settings of the automatically-launched fluidsynth process on RPi/Linux?

### DISCUSSION THREAD

**Discussion by GeoffreyPlitt** [FluidSynth/fluidsynth#1294] [General]
I'm running fluidsynth latest on Linux (Raspberry Pi 5, latest Raspbian Lite), with a USB MIDI controller.

When I start fluidsynth myself with `sudo fluidsynth -a alsa -c 2 -z 1024 -r 48000 -v -g 0.3 -o 'synth.polyphony=32' /usr/share/sounds/sf2/FluidR3_GM.sf2` and test by typing `noteon 0 60 120`,  the notes sound instant, nearly zero latency. ALSA rocks.

But when I connect a USB MIDI Controller (nanokey2), there's about 1/2 second latency. Later I realized my MIDI key presses weren't even going to the FluidSynth process I launched above. They went to one running somewhere in the background. Where is that configured?

I found it with `ps`:

$ ps auxw | grep fluid
pi         892  3.3  2.0 249528 170076 ?       SLsl 19:34   1:31 /usr/bin/fluidsynth -is /usr/share/sounds/sf3/default-GM.sf3

So I think all I need to do is change the command-line that runs this default/background FluidSynth process. Or prevent it from running and somehow connect the MIDI keyboard to Fluidsynth after I run fluidsynth myself, with ALSA. Which seems hard, because I think ALSA connections are best done before running FluidSynth, no?

To summarize:
1) Can I change the command-line used to run the system/background Fluidsynth process?
2) Or, can I prevent it from running, but somehow connect my MIDI keyboard to my own ALSA-ified Fluidsynth process after it's running?

**Comment by mawe42** [comment]
Raspberry Pi OS seems to use the debian fluidsynth package. And that comes with a systemd service to launch it on boot. So you should be able to prevent that instance from starting by disabling the service: `systemctl disable fluidsynth.service`

**Reply by mawe42** [reply]
Alternatively, you should have a `/etc/default/fluidsynth` file where you could change the parameters for the fluidsynth instance started from systemd.

**Comment by GeoffreyPlitt** [comment]
Wonderful, thanks!
